### PR TITLE
1st commit (from jjo/), with working implementation

### PR DESCRIPTION
**Pull request by jjo**
*  mark for later deletion those workloads (Deployments,
   StatefulSets, Jobs, Pods) lacking `--skip-labels` unless their
   namespace has them.
* deletion mark implemented as `kube-custodian.bitnami.com/expiration-mark`
  annotation
* `make test` shows 
```
[...]
PASS
coverage: 88.2% of statements
ok      github.com/bitnami-labs/kube-custodian/pkg/cleaner      (cached)        coverage: 88.2% of statements
=== RUN   Test_LabelsSubSet
--- PASS: Test_LabelsSubSet (0.00s)
PASS
coverage: 100.0% of statements
ok      github.com/bitnami-labs/kube-custodian/pkg/utils        (cached)        coverage: 100.0% of statements
```

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+
+ARG SRC_REPO=github.com/bitnami-labs/kube-custodian
+ARG SRC_TAG=master
+ARG ARCH=amd64
+
+RUN apk update && apk add git ca-certificates
+
+# Checkout specific version
+COPY . /go/src/${SRC_REPO}
+# RUN go get -d ${SRC_REPO}
```
Comment (jbianquetti-nami) [inline Dockerfile]: Would you remove the comments?

**Review by jbianquetti-nami — CHANGES_REQUESTED** [review]
I've compiled the binary from the branch source code and I get: 
```./bin/kube-custodian -v --namespace=default --dry-run run --tag-ttl 24h --required-labels created_by
ERRO[0000] rootCmd.Execute(): unknown flag: --required-labels
ERROR unknown flag: --required-labels 
```

As you can see, there's not a `--required-labels` option in this binary: 

```
./bin/kube-custodian --help
Cleanup stuff

Usage:
  kube-custodian [command]

Available Commands:
  run         Scan Kubernetes objects, mark for deletion (via annotation), delete those already "expired"
  version

Flags:
      --all-namespaces                 All namespaces
      --as string                      Username to impersonate for the operation
      --as-group stringArray           Group to impersonate for the operation, this flag can be repeated to specify multiple groups.
      --certificate-authority string   Path to a cert file for the certificate authority
      --client-certificate string      Path to a client certificate file for TLS
      --client-key string              Path to a client key file for TLS
      --cluster string                 The name of the kubeconfig cluster to use
      --context string                 The name of the kubeconfig context to use
      --dry-run                        Dry run
      --insecure-skip-tls-verify       If true, the server's certificate will not be checked for validity. This will make your HTTPS connections insecure
      --kubeconfig string              Path to a kube config. Only required if out-of-cluster
  -n, --namespace string               If present, the namespace scope for this CLI request
      --password string                Password for basic authentication to the API server
      --request-timeout string         The length of time to wait before giving up on a single server request. Non-zero values should contain a corresponding time unit (e.g. 1s, 2m, 3h). A value of zero means don't timeout requests. (default "0")
      --server string                  The address and port of the Kubernetes API server
      --token string                   Bearer token for authentication to the API server
      --user string                    The name of the kubeconfig user to use
      --username string                Username for basic authentication to the API server
  -v, --verbose count                  Increase verbosity
```

File: Dockerfile (hunk 1 above)
Comment (jjo) [reply Dockerfile]: Done.

**Comment by jjo** [comment]
Thanks+ @jbianquetti-nami for the review, PTAL.
 
> I've compiled the binary from the branch source code and I get:

It'd be `./bin/kube-custodian run --help`, I better clarified at `README.md`.

File: Dockerfile (hunk 1 above)
Comment (jjo) [reply Dockerfile]: Done.

**Comment by jjo** [comment]
@jbianquetti-nami BTW a `-v --dry-run` against our aws dev cluster (with some de-cluttering from DEBUG) -> https://gist.github.com/jjo/2957a5cc340924b579ad0390757906c2

**Review by jbianquetti-nami — APPROVED** [review]

**Comment by jbianquetti-nami** [comment]
OK, it's working now. SGTM.

**Comment by jbianquetti-nami** [comment]
FYI: cloud-custodian's guys have plans for this (https://groups.google.com/d/msg/cloud-custodian/eeL6xPKya1M/gxIdUw7jCAAJ) and have some bits already committed to the repository.

**Pull request merged**